### PR TITLE
Playsound improvements

### LINKED ISSUE #16510: Tank occupant sound weirdness
### Description

In some cases, the occupants of multitile vehicles seem to hear the outside from a source that doesn't align with where the tank actually is.

I'm not sure what the cause is because its not consistant, but in some cases it seems like their listening source gets left behind or something strange. Needs more investigation, mainly a note to self.

### Test Merges

NA

### Reproduction Steps

NA

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Playsound/explosion sounds are a bit more optimised. According to the RU lads it uses a lot less CPU so seemed worth doing.

Hearing sounds in tanks now gives proper directional sound, so this should:
Fix #16510

Little bit of code cleanup.

Note: This could be nicer but it turns out of 'clients_by_z lists are entirely nonfunctional, but I'll do that in a separate PR so its easier to look at.
## Why It's Good For The Game
Better perf, directional sound in tank good.
## Changelog
:cl:
qol: Sounds from outside tanks should now properly have directionality applied to them
code: Improved some sound code performance.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
DNM as the clients_by_z pr should be merged first, and will cut out a couple of for loops in this PR.

**Comment by Lumipharon** [comment]
this is good to review/merge now

File: code/controllers/subsystem/explosions.dm [hunk 1]
```diff
@@ -137,38 +137,45 @@ SUBSYSTEM_DEF(explosions)
...
-						M.playsound_local(epicenter, creak_sound, far_volume*3, 1, frequency, falloff = 5)//ship groaning under explosion effect
-					if(baseshakeamount > 0)
-						shake_camera(M, 7, clamp(baseshakeamount*0.15, 0, 1.5))
+		if(devastation_range)
+			explosion_sound = SFX_EXPLOSION_LARGE
+		else if(heavy_impact_range)
+			explosion_sound = SFX_EXPLOSION_MED
+		else if(light_impact_range || weak_impact_range)
+			explosion_sound = SFX_EXPLOSION_SMALL
+			far_explosion_sound = SFX_EXPLOSION_SMALL_DISTANT
```
Comment (TiviPlus) [inline code/controllers/subsystem/explosions.dm:146]: sidenote can we optimize this even more by avoiding the lookups for this in the get_sfx giant elif chain? ie just call getsfx once here and play the same sound for everyone instead off lookup individually

**Review by TiviPlus — APPROVED** [review]

File: code/controllers/subsystem/explosions.dm (hunk 1 above)
Comment (Lumipharon) [reply code/controllers/subsystem/explosions.dm:146]: done

**Pull request merged**